pandas/core/indexes/numeric.py

```diff
@@ -56,8 +57,16 @@ def __new__(cls, data=None, dtype=None, copy=False, name=None, fastpath=None):
             if fastpath:
                 return cls._simple_new(data, name=name)
 
-        # is_scalar, generators handled in coerce_to_ndarray
-        data = cls._coerce_to_ndarray(data)
+        # Coerce to ndarray if not already ndarray or Index
+        if not isinstance(data, (np.ndarray, Index)):
+            if is_scalar(data):
+                raise cls._scalar_data_error(data)
+
+            # other iterable of some kind
+            if not isinstance(data, (ABCSeries, list, tuple)):
+                data = list(data)
+
+            data = np.asarray(data, dtype=dtype)
```

Comment:
Maybe add a line like `dtype = dtype or cls._default_dtype` prior to this to address my previous comment?  Or could even just have `dtype=cls._default_dtype` here for brevity, as this should be the only thing allowed after #29545 is merged.